### ISSUE TITLE
how to save the loss during lora finetune

### ISSUE THREAD

**Issue by Tan-Junwen** [ostris/ai-toolkit#218]
## This is for bugs only

Did you already ask [in the discord](https://discord.gg/VXmU2f5WEU)?

Yes/No

You verified that this is a bug and not a feature request or question by asking [in the discord](https://discord.gg/VXmU2f5WEU)?

Yes/No

**Comment by Poseidon-fan** [comment]
The `loss_dict` is in the `run` method of class `BaseSDTrainProcess` in the file `jobs/process/BaseSDTrainProcess.py`:

~~~python
            # flush()
            ### HOOK ###
            loss_dict = self.hook_train_loop(batch_list)
~~~

You can declare a global list and record the loss history:
~~~python
...
loss_history = []  # global variable


...
            # flush()
            ### HOOK ###
            loss_dict = self.hook_train_loop(batch_list)
            loss_history.append(loss_dict['loss'])

...
        ###################################################################
        ##  END TRAIN LOOP
        ###################################################################

        plt.figure(figsize=(8, 4))
        plt.plot(loss_history, marker='o', linestyle='-', label='Data')
        plt.title('Line Chart of Float List')
        plt.xlabel('step')
        plt.ylabel('loss')
        plt.grid(True)
        plt.legend()
        plt.savefig('loss.png')

        self.progress_bar.close()

...
~~~
